src/partition/repart.c

```diff
@@ -373,6 +373,7 @@ typedef struct Partition {
         MinimizeMode minimize;
         uint64_t verity_data_block_size;
         uint64_t verity_hash_block_size;
+        uint64_t verity_max_data_size;
```

Comment:
why do we need this? why not derive this automatically from the max size of the data partition to protect? why would we want this to deviate?

Reply:
Good point, haven't thought of reusing that property, this would make some of my introduced assertions unnecessary.

So we'd only derive the default size of the hash partition from `SizeMaxBytes=` on the data sibling.
When `SizeMinBytes=` is defined for the verity hash partition that would overrule the derived default and an error would be returned if the hash partition's `SizeMinBytes=`/`SizeMaxBytes=` are less than the calculated value?

Reply:
well, i'd probably warn in that case only, not fail. I mean, if people explicitly want some size limits and they don't make much sense that's certainly something to mention, but dunno, maybe people want to test something, hence i wouldn't refuse.

Reply:
updated this to use `SizeMaxBytes=` from the data sibling instead of a new property.
Now a warning is logged when `SizeMaxBytes=` of the hash partition is smaller than the calculated size required to store the hash data for `SizeMaxBytes=` of its data sibling.

I'm not sure about setting both `size_min` and `size_max` to the calculated size. Maybe `size_min` should only be set if it is not configured by the user?

Reply:
> I'm not sure about setting both size_min and size_max to the calculated size. Maybe size_min should only be set if it is not configured by the user?

@poettering so I'm thinking now that both `size_min` and `size_max` should only be set to the calculated value when they are not already configured, meaning the user would still be able to configure a larger size for the hash partition than the one implied by `SizeMaxBytes=` from the data sibling. Not sure why one need that though

Reply:
Okay, updated the code to only set `min_size` when it's not yet defined and dropped `max_size` so that user-defined sizes are not overridden.